s3/s3.go

```diff
@@ -0,0 +1,243 @@
+package s3
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	awsS3 "github.com/aws/aws-sdk-go/service/s3"
+	"github.com/taskcluster/taskcluster-cli/config"
+	"github.com/taskcluster/taskcluster-cli/extpoints"
+	"github.com/taskcluster/taskcluster-client-go"
+	"github.com/taskcluster/taskcluster-client-go/auth"
+)
+
+func init() {
+	extpoints.Register("s3", S3{})
+}
+
+type S3 struct{}
+
+func (S3) ConfigOptions() map[string]extpoints.ConfigOption {
+	return nil
+}
+
+func (S3) Summary() string {
+	return "Uploads/downloads file to/from AWS S3."
+}
+
+func usage() string {
+	return `Upload/Download file to/from AWS S3.
+Usage:
+  taskcluster S3 put <file> <bucket> <prefix>
+  taskcluster S3 get <key> <bucket> <prefix>
```

Comment:
What is the difference between `<key>` and `<prefix>` both are addresses in s3, right?

I would think... `taskcluster S3 get <bucket> <prefix> [target]` might be better...
If `[target]` isn't given then `filepath.Join(os.Getwd(), path.Basename(prefix))` is used as filename...